### PR TITLE
Add radio and segment input

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/components/RadioInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        i: O,
+    ) => RRP = React.useCallback((key: V, item: O) => {
+        // NOTE: this is required becase this is advance usecase
+        // and the typings
+        const radioProps: Pick<RRP, 'inputName' | 'label' | 'name' | 'onClick' | 'value'> = {
+            inputName: name,
+            label: radioLabelSelector(item),
+            name: key,
+            onClick: handleRadioClick,
+            value: key === value,
```
Comment (AdityaKhatri) [inline src/components/RadioInput/index.tsx:78]: Pass disabled and disable the input

Comment (frozenhelium) [reply src/components/RadioInput/index.tsx:78]: Done!

**Comment by AdityaKhatri** [comment]
Add readonly

**Comment by frozenhelium** [comment]
@AdityaKhatri done!

**Review by AdityaKhatri — APPROVED** [review]

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -36,9 +36,10 @@ export interface Props<N> extends Omit<
     big?: boolean;
     name: N;
     onClick?: (name: N, e: React.MouseEvent<HTMLButtonElement>) => void;
+    readOnly?: boolean;
```
Comment (tnagorra) [inline src/components/Button/index.tsx:39]: Button should not have a readOnly field.
What does this do?

File: src/stories/RadioInput.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            onChange={setValue}
+        />
+    );
+};
+
+export const Default = Template.bind({});
+Default.args = {
+    name: 'test',
+    value: '1',
+};
```
Comment (tnagorra) [inline src/stories/RadioInput.stories.tsx:50]: Please add more stories.

File: src/stories/SegmentInput.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+Default.args = {
+    name: 'test',
+    value: '1',
+    segmentRendererParams: (o) => {
+        if (o.key === '1') {
+            return { icons: <FaApple /> };
+        }
+
+        return {};
+    },
```
Comment (tnagorra) [inline src/stories/SegmentInput.stories.tsx:58]: Please add more stories

**Review by subinasr — APPROVED** [review]

File: src/stories/RadioInput.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    value: '1',
+    disabled: true,
+};
+
+export const ReadOnly = Template.bind({});
+ReadOnly.args = {
+    name: 'test',
+    value: '1',
+    readOnly: true,
+};
```
Comment (tnagorra) [inline src/stories/RadioInput.stories.tsx:64]: Please also add stories to show error and hint

File: src/stories/SegmentInput.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    value: '1',
+    readOnly: true,
+    segmentRendererParams: (o) => {
+        if (o.key === '1') {
+            return { icons: <FaApple /> };
+        }
+
+        return {};
+    },
+};
```
Comment (tnagorra) [inline src/stories/SegmentInput.stories.tsx:87]: Please also add stories to show error and hint

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**